### PR TITLE
Add {fork_name}_enabled functions

### PR DESCRIPTION
**Pull request by eserilev**
Add  `{fork_name}_enabled` functions to the `ForkName` impl and use them in place of `fork_name >= ForkName::{fork_name_variant}` checks and in places where we can remove `match` boilerplate code. 

There was some discussion [here](https://github.com/sigp/lighthouse/pull/5940#discussion_r1644693373) about implementing something like this to help reduce chances of developer error.

### CODE REVIEW

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 1]
```diff
@@ -2574,10 +2574,11 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
...
         let current_fork = self.spec.fork_name_at_slot::<T::EthSpec>(self.slot()?);
-        if let ForkName::Base | ForkName::Altair | ForkName::Bellatrix = current_fork {
-            Ok(false)
-        } else {
+
+        if current_fork.capella_enabled() {
             Ok(true)
+        } else {
+            Ok(false)
         }
```
Comment (dapplion) [inline beacon_node/beacon_chain/src/beacon_chain.rs]: Just return `Ok(current_fork.capella_enabled())`

**Review by dapplion — COMMENTED** [review]
Should we name if `{fork_name}_enabled` or `after_{fork_name}`?

**Comment by eserilev** [comment]
im open to either `{fork_name}_enabled` or `after_{fork_name}`, whatever makes the most sense

File: beacon_node/beacon_chain/src/beacon_chain.rs (hunk 1 above)
Comment (eserilev) [reply beacon_node/beacon_chain/src/beacon_chain.rs]: 3ba2f78

**Comment by dapplion** [comment]
Let's do `{fork_name}_enabled` we already committed to it

**Comment by realbigsean** [comment]
> Let's do `{fork_name}_enabled` we already committed to it

We get autocomplete for the right fork faster this way :)

**Review by dapplion — APPROVED** [review]
All logic is equivalent 👍 

This is such a nice cosmetic improvement, no need to sit on this one for longer.

@eserilev please fix conflicts, and good to go

**Comment by eserilev** [comment]
conflicts are resolved. Note that I've added an additional electra related TODO in an ef test here: 
6f94f62

**Review by dapplion — APPROVED** [review]
LGTM

**Comment by jimmygchen** [comment]
@eserilev would you mind fixing the conflicts again, and lets get this merged!

**Review by jimmygchen — APPROVED** [review]

**Comment by jimmygchen** [comment]
@mergify queue

**Pull request merged**